apiserver/facades/agent/caasapplication/application_test.go

```diff
@@ -35,11 +36,16 @@ type CAASApplicationSuite struct {
 	st         *mockState
 	clock      *testclock.Clock
 	broker     *mockBroker
+
+	controllerConfigGetter *MockControllerConfigGetter
 }
 
 func (s *CAASApplicationSuite) SetUpTest(c *gc.C) {
 	s.BaseSuite.SetUpTest(c)
 
+	ctrl := gomock.NewController(c)
+	s.controllerConfigGetter = NewMockControllerConfigGetter(ctrl)
```

Comment:
why is this mock never called, (i.e. some test method should be calling `s.controllerConfigGetter.ControllerConfig(context.Background())`), if there are no calls it means you don't need a mock just nil

Reply:
So indeed, in the tests we never call the controller config getter so you can easily pass a `nil` instead of a mock. Or ideally, add a test that does call the getter